p2panda-net-next/src/actors/gossip/receiver.rs

```diff
@@ -20,36 +21,31 @@ pub enum ToGossipReceiver {
 
 pub struct GossipReceiverState {
     receiver: Option<IrohGossipReceiver>,
+    session_ref: ActorRef<ToGossipSession>,
 }
 
-pub struct GossipReceiver {
-    session: ActorRef<ToGossipSession>,
-}
-
-impl GossipReceiver {
-    pub fn new(session: ActorRef<ToGossipSession>) -> Self {
-        Self { session }
-    }
-}
+#[derive(Default)]
+pub struct GossipReceiver;
```

Comment:
To use `ThreadLocalActor` `Self` needs to implement `Default`, this is why I've needed to move the `ActorRef` into the state instead, which also feels like a good place.